src/gifwrite.c

```diff
@@ -12,15 +12,58 @@ void generate_GIF_data (struct context * context) {
   if ((uint8_t) (depth >> 8) > overall) overall = depth >> 8;
   if ((uint8_t) (depth >> 16) > overall) overall = depth >> 16;
   if (overall > 8) overall = 8;
-  header[4] = (overall - 1) << 4;
-  header[5] = header[6] = 0;
+  bytewrite(header + 4, (overall - 1) << 4, 0, 0);
   if (context -> source -> palette)
     generate_GIF_data_with_palette(context, header);
   else
     generate_GIF_data_from_raw(context, header);
   byteoutput(context, 0x3b);
 }
 
+#define checkboundaries(buffertype, buffer, frame) do {                                                                          \
+  size_t index = 0;                                                                                                              \
+  if (boundaries) {                                                                                                              \
+    while (index < context -> source -> width * boundaries[frame].top) if (buffer[index ++] != transparent) goto findbounds;     \
+    for (uint_fast16_t row = 0; row < boundaries[frame].height; row ++) {                                                        \
+      for (uint_fast16_t col = 0; col < boundaries[frame].left; col ++) if (buffer[index ++] != transparent) goto findbounds;    \
+      index += boundaries[frame].width;                                                                                          \
+      for (uint_fast16_t col = boundaries[frame].left + boundaries[frame].width; col < context -> source -> width; col ++)       \
+        if (buffer[index ++] != transparent) goto findbounds;                                                                    \
+    }                                                                                                                            \
+    while (index < framesize) if (buffer[index ++] != transparent) goto findbounds;                                              \
+    left = boundaries[frame].left;                                                                                               \
+    top = boundaries[frame].top;                                                                                                 \
+    width = boundaries[frame].width;                                                                                             \
+    height = boundaries[frame].height;                                                                                           \
+  } else {                                                                                                                       \
```

Comment:
This `else` takes the place of an unnecessary `goto`.

This control flow:

```c
if (boundaries) {
  if (various stuff) goto findbounds;
  goto gotbounds;
}
findbounds:
more stuff;
gotbounds:
```

was refactored to this:

```c
if (boundaries) {
  if (various stuff) goto findbounds;
} else {
  findbounds:
  more stuff;
}
```